tests/app/main/views/test_services.py

```diff
@@ -574,6 +574,56 @@ def test_service_update_with_one_service_feature(self, data_api_client):
         assert response.status_code == 302
 
     @mock.patch('app.main.views.services.data_api_client')
+    def test_service_update_with_multiquestion_validation_error(self, data_api_client):
+        data_api_client.get_service.return_value = {'services': {
+            'id': 1,
+            'supplierId': 2,
+            'frameworkSlug': 'g-cloud-9',
+            'lot': 'cloud-hosting',
+            'serviceFeatures': [
+                "bar",
+            ],
+            'serviceBenefits': [
+                "foo",
+            ],
+        }}
+        mock_api_error = mock.Mock(status_code=400)
+        mock_api_error.json.return_value = {
+            "error": {"serviceBenefits": "under_10_words", "serviceFeatures": "under_10_words"}
+        }
+        data_api_client.update_service.side_effect = HTTPError(mock_api_error)
+
+        response = self.client.post(
+            '/admin/services/1/edit/service-features-and-benefits',
+            data={
+                'serviceFeatures': 'one 2 three 4 five 6 seven 8 nine 10 eleven',
+                'serviceBenefits': '11 10 9 8 7 6 5 4 3 2 1',
+            },
+            follow_redirects=True
+        )
+        assert data_api_client.update_service.call_args_list == [(
+            (
+                '1',
+                {
+                    'serviceFeatures': ['one 2 three 4 five 6 seven 8 nine 10 eleven'],
+                    'serviceBenefits': ['11 10 9 8 7 6 5 4 3 2 1'],
+                },
+                'test@example.com',
+            ),
+            {},
+        )]
+
+        document = html.fromstring(response.get_data(as_text=True))
+
+        validation_banner = document.xpath("//div[@class='validation-masthead']//text()[normalize-space(.) != '']")
+        validation_banner_text = ' '.join([elem.strip() for elem in validation_banner])
+        validation_errors = [error.strip() for error in document.xpath("//span[@class='validation-message']//text()")]
+        assert response.status_code == 400
+        assert validation_banner_text == "There was a problem with your answer to: Service benefits Service features"
+        assert "You can’t write more than 10 words for each feature." in validation_errors
+        assert "You can’t write more than 10 words for each benefit." in validation_errors
```

Comment:
I think we can be more exact in asserting both the masthead (with its bookmark links) and the errors. Something like:

```
validation_banner_h1 = document.xpath("//h1[@class='validation-masthead-heading']//text()")[0].strip()
assert validation_banner_h1 == "There was a problem with your answer to:"

validation_banner_links = [
    (anchor.text_content(), anchor.get('href')) for anchor in document.xpath("//a[@class='validation-masthead-link']")
]
assert validation_banner_links == [
    ("Service benefits", "#serviceBenefits"),
    ("Service features", "#serviceFeatures")
]

validation_errors = [error.strip() for error in document.xpath("//span[@class='validation-message']//text()")]
assert validation_errors == [
    "You can’t write more than 10 words for each feature.",
    "You can’t write more than 10 words for each benefit."
]
```

Going to the effort of doing this revealed to me that a) there are two h1 elements in the page, b) the list of bookmark links in the masthead is in a different order to the questions! I think the latter's a known bug, but I had forgotten about it until just now...

Reply:
That rings a bell.  It might be a problem that was fixed in supplier app, but I'm not sure, will check it out.

Reply:
Many thanks for not making vague suggestions BTW 😸

Reply:
I looked at the ordering thing but supplier app doesn't deal with it that I can see.  I think we leave it as-is for now.